src/lib/fauna-client.mjs

```diff
@@ -90,3 +70,88 @@ export default class FaunaClient {
     return undefined;
   }
 }
+
+export const getSecret = async () => {
+  const credentials = container.resolve("credentials");
+  if (!credentials.databaseKeys.key) {
+    return await credentials.databaseKeys.getOrRefreshKey();
+  }
+  return credentials.databaseKeys.key;
+}
+
+export const retryInvalidCredsOnce = async (initialSecret, fn) => {
+  try {
+    return await fn(initialSecret);
+  } catch (err) {
+    // If it's a 401, we need to refresh the secret. Let's just do type narrowing here
+    // vs doing another v4 vs v10 check.
+    if (err && (err.httpStatus === 401 || err.requestResult?.statusCode === 401)) {
+      const credentials = container.resolve("credentials");
+
+      await credentials.databaseKeys.onInvalidCreds();
+      const refreshedSecret = await credentials.databaseKeys.getOrRefreshKey();
```

Comment:
Since we already awaited the refresh via onInvalidCreds, safe to just access ".key" here

Reply:
I'll clean this up in a follow-up.